### PR TITLE
pacific: mgr/dashboard: test prometheus rules through promtool

### PR DESCRIPTION
**Pull request by aaSharma14**
backport tracker: https://tracker.ceph.com/issues/49658

---

backport of

* https://github.com/ceph/ceph/pull/39246
* https://github.com/ceph/ceph/pull/40205 (one commit)
* https://github.com/ceph/ceph/pull/40300
* https://github.com/ceph/ceph/pull/40395

parent tracker: https://tracker.ceph.com/issues/45415

this backport was staged using ceph-backport.sh version 16.0.0.6848
find the latest version at https://github.com/ceph/ceph/blob/master/src/script/ceph-backport.sh

### CODE REVIEW

**Review by idryomov — DISMISSED** [review]
Please drop unrelated libzbd and boost changes.

**Comment by aaSharma14** [comment]
> Please drop unrelated libzbd and boost changes.

@tchaikov Shouldn't these changes be in pacific branch?

**Comment by aaSharma14** [comment]
jenkins test api

**Comment by aaSharma14** [comment]
jenkins retest this please

**Comment by tchaikov** [comment]
> > Please drop unrelated libzbd and boost changes.
> 
> @tchaikov Shouldn't these changes be in pacific branch?

@aaSharma14 i don't think so. if you think otherwise, could you help me understand why?

**Comment by aaSharma14** [comment]
> > > Please drop unrelated libzbd and boost changes.
> > 
> > 
> > @tchaikov Shouldn't these changes be in pacific branch?
> 
> @aaSharma14 i don't think so. if you think otherwise, could you help me understand why?

I thought they would be required for running promtool without docker. Anyways I have removed those changes from all the three backport PRs. Thanks

File: install-deps.sh [hunk 1]
```diff
@@ -344,6 +344,9 @@ else
         case "$ID" in
             fedora)
                 $SUDO dnf install -y dnf-utils
+                $SUDO dnf install -y docker-ce docker-ce-cli containerd.io
```
Comment (tchaikov) [inline install-deps.sh:347]: could you please include e33e3a931db97d01318643ec686fe63fdd614082 in this backport?

File: monitoring/prometheus/alerts/test_alerts.yml [hunk 2]
```diff
@@ -0,0 +1,774 @@
+rule_files:
```
Comment (tchaikov) [inline monitoring/prometheus/alerts/test_alerts.yml:1]: could you please include 8d2f39e6c568afb6880689160212bcc93057e194 in this backport as well?

File: install-deps.sh (hunk 1 above)
Comment (aaSharma14) [reply install-deps.sh]: Thanks @tchaikov . Done.

File: monitoring/prometheus/alerts/test_alerts.yml (hunk 2 above)
Comment (aaSharma14) [reply monitoring/prometheus/alerts/test_alerts.yml:1]: Done.

**Review by idryomov — COMMENTED** [review]
My concern has been addressed, deferring to @tchaikov and the Dashboard team on the backport itself.

**Comment by smithfarm** [comment]
Edited the PR description to clarify which PRs/commits are being backported here.

**Comment by avanthakkar** [comment]
jenkins test docs

**Review by avanthakkar — APPROVED** [review]

**Review by nizamial09 — APPROVED** [review]

**Review by alfonsomthd — APPROVED** [review]

**Pull request merged**